libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailDialog/ScheduledMailDialogRenderer/Attachments/Attachments.tsx

```diff
@@ -0,0 +1,97 @@
+import identity from "lodash/identity";
+import { ITheme, IInsightWidget } from "@gooddata/sdk-backend-spi";
+import { withTheme } from "@gooddata/sdk-ui-theme-provider";
+
+import { AttachmentsSelectionDropdown } from "./AttachmentsSelectionDropdown";
+import { FormatOptionsDropdown } from "./FormatOptionsDropdown";
+import { IWidgetExportConfiguration, IWidgetsSelection } from "../../interfaces";
+import { objRefToString } from "@gooddata/sdk-model";
+
+export interface IAttachmentsOwnProps {
+    theme?: ITheme;
+    dashboardTitle: string;
+    insightWidgets: IInsightWidget[];
+    dashboardSelected: boolean;
+    widgetsSelected: IWidgetsSelection;
+    configuration: IWidgetExportConfiguration;
+    canExportReport?: boolean;
+    onAttachmentsSelectionChanged(dashboardSelected: boolean, widgetsSelected: IWidgetsSelection): void;
+    onAttachmentsConfigurationChanged(configuration: IWidgetExportConfiguration): void;
+}
+
+const AttachmentItem: React.FC<{ format: string }> = ({ format, children }) => (
+    <div className="gd-dashboard-attachment">
+        <span className="gd-dashboard-attachment-format">{format}</span>
+        <span className="gd-dashboard-attachment-name">{children}</span>
+    </div>
+);
+
+export type IAttachmentsProps = IAttachmentsOwnProps & WrappedComponentProps;
+
+const AttachmentsComponent = (props: IAttachmentsProps) => {
+    const {
+        dashboardTitle,
+        dashboardSelected,
+        widgetsSelected,
+        insightWidgets = [],
+        configuration,
+        canExportReport,
+        onAttachmentsSelectionChanged,
+        onAttachmentsConfigurationChanged,
+    } = props;
+
+    const isSomeWidgetSelected = Object.values(widgetsSelected).some(identity);
+    const selectedWidgetsTitles = insightWidgets
+        .filter((widget) => widgetsSelected[objRefToString(widget)])
+        .map((widget) => widget.title);
+
+    return (
+        <div className="gd-input-component gd-schedule-email-attachments s-schedule-email-attachments">
+            <label className="gd-label">
+                <FormattedMessage id="dialogs.schedule.email.attachments.label" />
+            </label>
+            <div className="gd-dashboard-attachment-list">
+                <div className="gd-dashboard-attachment-list-content">
+                    {dashboardSelected && (
+                        <AttachmentItem format="PDF">
+                            <span className="shortened-name">{dashboardTitle}</span>
+                        </AttachmentItem>
+                    )}
+                    {selectedWidgetsTitles.length !== 0 && (
+                        <AttachmentItem format={configuration.format}>
+                            <span className="shortened-name">{selectedWidgetsTitles.join(", ")}</span>
+                            <span>
+                                {selectedWidgetsTitles.length > 1 && `(${selectedWidgetsTitles.length})`}
+                            </span>
```

Comment:
Is there some reason why the span can render empty text? If not, I would move the <span> into the condition.